myuw/dao/instructor_schedule.py

```diff
@@ -155,17 +164,59 @@ def is_instructor(request):
     Determines if user is an instructor of the request's term
     """
     try:
-        person = get_person_of_current_user()
         term = get_current_quarter(request)
-        sections = _get_instructor_sections(person, term)
-        return (len(sections) > 0)
+        user = get_user_model()
+        qset = SeenInstructor.objects.filter(user=user)
+        if qset.count() > 0:
+            return True
+
+        person = get_person_of_current_user()
+        sections = _get_instructor_sections(person, term, future_terms=2)
+        if len(sections) > 0:
+            SeenInstructor.objects.update_or_create(
+                user=user,
+                year=term.year,
+                quarter=term.quarter)
+            return True
+
+        return False
     except DataFailureException as err:
         if err.status == 404:
             return False
 
         raise
 
 
+def get_prior_instructed_terms(term):
+    terms = []
+    prior_years = getattr(settings, "MYUW_PRIOR_INSTRUCTED_TERM_YEARS",
+                          MYUW_PRIOR_INSTRUCTED_TERM_YEARS_DEFAULT)
+    for i in range(prior_years * 4):
+        try:
+            term = get_term_before(term)
+            terms.insert(0, term)
+        except DataFailureException as ex:
+            if ex.status == 404:
+                pass
+
+    return terms
+
+
+def get_future_instructed_terms(term):
```

Comment:
holding off on this since in the instructed case there are specific requirements for past and future instructed quarter spans.  let's add consolidating to the backlog